### DISCUSSION TITLE
MongoDB - A number used in the query does not fit into a 64 bit signed integer.

### DISCUSSION THREAD

**Discussion by kaykhan** [prisma/prisma#15251] [Q&A] [answered]
I am trying to insert a very large number into MongoDB and im having difficulty determine what supported datatype i can use in prisma. 

When i attempt to store a value i get the following error:

>   Query parsing failure: A number used in the query does not fit into a 64 bit signed integer. Consider using `BigInt` as field type if you're trying to store large integers.

**schema**

```
model SalesByContractAddress {
    price                  BigInt?
   ...
}
```

**data**

```
price: 21049200000000000000,
```

This price value is ethereum price formatted with the smallest denomination. I need to able to insert this into my mongodb. 

What prisma datatype can i use here to store that vlaue

**Comment by austincrim** [comment, marked as answer]
Hey @kaykhan 👋,

The given number is too large for the [MongoDB `Long` type](https://mongodb.github.io/node-mongodb-native/3.5/api/Long.html) to handle (which is what `BigInt` translates to on MongoDB).

You could try storing the value as a string instead.

**Reply by kaykhan** [reply]
Yes i eventually stored it as a string becuase prisma does not support mongodb decimal128 data type.
